src/proxy.sol

```diff
@@ -84,10 +91,23 @@ contract Proxy is TitleOwned {
+    bytes32 proxyCodeHash;
+
     event Created(address indexed sender, address indexed owner, address proxy, uint tokenId);
-    mapping (uint => address) public proxies;
+
+    function proxies(uint accessToken) public view returns(address) {
+        // create2 address calculation
+        // keccak256(0xff ++ deployingAddr ++ salt ++ keccak256(bytecode))[12:]
+        bytes32 _data = keccak256(
+            abi.encodePacked(bytes1(0xff), address(this), keccak256(abi.encodePacked(accessToken)), proxyCodeHash)
+        );
+        return address(bytes20(_data << 96));
+    }
 
     constructor() Title("Tinlake Actions Access Token", "TAAT") public {
+        proxyCodeHash = keccak256(abi.encodePacked(proxyCode));
```

Comment:
if the bytecode is hardcoded, I think we should hardcode this as well.

Reply:
On second thought, why don't we pass the contract bytecode into the constructor?

Reply:
you can ensure it is the correct hash if you do it in the constructor